### ISSUE TITLE
RFE: Allow `LIKE` for binary types

### ISSUE THREAD

**Issue by ensc** [diesel-rs/diesel#3127]
### Versions

- **Diesel:**  master
- **Database:**  postgres

## Problem Description

`LIKE` and related operators (https://github.com/diesel-rs/diesel/blob/master/diesel/src/expression_methods/text_expression_methods.rs) can be applied to binary types.  E.g. I have a table where the filenames are of `BYTEA` type:

```
\d file
 fname  | bytea                       |           | not null | 
```

Queries like

```
# select count(*) from file where fname like '%.tgz';
 count  
--------
 107889
```

are working fine and I would like to use them in diesel.

In stable branch, it was possible to implement this manually by something like

```rust
diesel::expression::operators::Like::new(self.as_expression(), other.as_expression())
```

But `expressions::operators` became non-public recently and expression above fails to compile now.

**Comment by weiznich** [comment]
First of all: I do not consider that to be a bug. This should likely be filled as feature request, so a [thread in our discussion forum.](https://github.com/diesel-rs/diesel/discussions/new?category=ideas)  https://github.com/diesel-rs/diesel/discussions/3039 seems to be related.

The corresponding item was never part of the public API, which means we did not guarantee anything there. That written: I can totally understand that this might break existing users, and that you want to have a replacement. So as a workaround you can just use [`infix_operator!()`](https://docs.diesel.rs/master/diesel/macro.infix_operator.html) to define the corresponding operator in your crate. Other than that I might be fine with adding official support for this to diesel, if someone can bring up some documentation for this operator on the postgres side + fills a PR that adds support (+ some tests).

**Comment by weiznich** [comment]
Closed by #3150
